feat(atomic): added atomic-notifications

btaillon opened the pull request:
https://coveord.atlassian.net/browse/KIT-1923

<img width="1519" alt="image" src="https://user-images.githubusercontent.com/54454747/183511181-a32898fa-5e9a-4c74-af3f-3231f83317ad.png">

I made it only display the first notification returned by the API. In your opinions, do you think it should be able to display multiple notifications? If so, I'll rename it to `atomic-notifications`. I'm going to wait for Johnny's response when he's back.

Missing:
* Support for links (suggested by Johnny)
* Allowing atomic-aria-live to read changes that happened at the same time.

jpmarceau commented on packages/atomic/src/components/search/atomic-notification/atomic-notification.tsx:
```diff
@@ -0,0 +1,85 @@
+import {Heading} from '../../common/heading';
+
+/**
+ * The `atomic-notification` component is responsible for displaying notifications generated by the Coveo Search API (see [Trigger](https://docs.coveo.com/en/1458/tune-relevance/trigger-query-pipeline-feature)).
```
```suggestion
 * The `atomic-notification` component is responsible for displaying notifications generated by the Coveo Search API (see [Trigger](https://docs.coveo.com/en/1458)).
```
(everything after the ID is unnecessary)

jpmarceau reviewed (approved):
Excellent strings, as usual 👌

ThibodeauJF commented on packages/atomic/src/pages/index.html:
```diff
@@ -211,6 +211,9 @@
             </atomic-sort-dropdown>
             <atomic-did-you-mean></atomic-did-you-mean>
           </atomic-layout-section>
+          <atomic-layout-section section="notification"
```
I'm not entirely convinced a new section is needed especially since there's only this component inside (requires a doc update + our customer to implement it), it could easily fit in either results or status.

btaillon replied:
If it doesn't have its own section, then I'll add it in `status`.

olamothe replied:
Indeed, adding sections has a "cost" (doc, maintenance, burden on implementers). 

It needs to be really worth it before we add a new one.

ThibodeauJF reviewed (approved):
Nice!

olamothe commented on packages/atomic/src/components/search/atomic-notification/atomic-notification.tsx:
```diff
@@ -0,0 +1,85 @@
+        class="shadow-lg bg-background border border-neutral-dark rounded-md p-6 flex items-center"
+      >
+        <atomic-icon
+          icon={InfoIcon}
```
Suggestion: Make this icon a prop on the component, with the default value being the `InfoIcon`. It would be nice if there was also a way to say that you want "no icon" at all.

olamothe reviewed (approved):
Great PR with docs, tests, storybook... 😎 

GG !

btaillon commented:
Re-requesting reviews because I added support for icons and renamed the component to `atomic-notifications` (despite only showing 1 notification at a time)
